python/foglamp/services/south/server.py

```diff
@@ -244,4 +261,10 @@ def run(self):
         new_handle = self._plugin.plugin_reconfigure(self._plugin_handle, new_config)
         self._plugin_handle = new_handle
 
+        # Executes the requested plugin type with new config
+        if self._plugin_info['mode'] == 'async':
+            await  self._exec_plugin_async(new_config)
+        elif self._plugin_info['mode'] == 'poll':
+            asyncio.ensure_future(self._exec_plugin_poll(new_config))
+
```

Comment:
I think restart is responsibility of reconfigure, how it works for CoAP/ HTTP south; if no other possibilty .. add a docstring with FIXME?